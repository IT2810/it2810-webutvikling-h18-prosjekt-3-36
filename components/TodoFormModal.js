import React from 'react';
import {
  Image,
  Platform,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  Alert,
  Button,
  TextInput,
  SafeAreaView
} from 'react-native';
import Modal from "react-native-modal";
import DatePicker from 'react-native-datepicker';
import { FloatingAction } from 'react-native-floating-action';

export default class TodoForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {isModalVisible: false}
  }

  _toggleModal = () => {
    this.setState({ text: null });
    this.setState({ date: null });
    // clear the text and date aswell as to toggle whether the modal is visible or not.
    this.setState({ isModalVisible: !this.state.isModalVisible });
  }

  addNewTodo = () => {
    // Kjør addNewTodo metoden til TodoScreen så alt av todo states blir gjort av den.
    this.props.addNewTodo(this.state.text, this.state.date, this.state.categoryChosen);
    this._toggleModal();
  }

  render() {
    // actions er hvilken handlinger du kan gjøre etter du har trykker på den store + knappen. Vi har valgt å ha kategorier her. 
    const actions = [{
      text: 'Work',
      name: 'work',
      color: "red",
      position: 1
    }, {
      text: 'School',
      name: 'school',
      color: "blue",
      position: 2
    }, {
      text: 'Fun',
      name: 'fun',
      color: "orange",
      position: 3
    }, {
      text: 'Other',
      name: 'other',
      color: "green",
      position: 4
    }];
  return (
    <SafeAreaView style={styles.container}>
      <Modal isVisible={this.state.isModalVisible}>
        <View style={{ flex: 1, backgroundColor: "white" }}>
          <View style={styles.buttonContainer}>
            <View style={styles.button} ><Button title="Cancel" onPress={this._toggleModal} color="red" style={styles.button}></Button></View>
            <View style={styles.button} ><Button title="Submit" onPress={this.addNewTodo} color="green" style={styles.button}></Button></View>
          </View>
          <TextInput
            editable={true}
            multiline
            onChangeText={(text) => this.setState({ text })}
            value={this.state.text}
            placeholder="Write something..."
            placeholderTextColor="grey"
            style={styles.textInput}
            underlineColorAndroid='rgba(0,0,0,0)'
            onSubmitEditing={this.addNewTodo}
            autoFocus={true}></TextInput>
          <DatePicker
            style={{ width: 200 }}
            date={this.state.date}
            mode="date"
            placeholder="select date"
            format="YYYY-MM-DD"
            minDate="1990-01-01"
            maxDate="2025-01-01"
            confirmBtnText="Confirm"
            cancelBtnText="Cancel"
            customStyles={{
              dateIcon: {
                position: 'absolute',
                left: 0,
                top: 4,
                marginLeft: 0
              },
              dateInput: {
                marginLeft: 36
              }
            }}
            onDateChange={(date) => { this.setState({ date: date }) }}
          />
          <Text>Add a date for your todo (optional)</Text>
        </View>
      </Modal>
      <View style={styles.container}>
        <FloatingAction
          actions={actions}
          position="right"
          onPressItem={
            (name) => {
              this.setState({ isModalVisible: true });
              this.setState({ categoryChosen: name });
            }
          }
        />
      </View>
    </SafeAreaView>
  );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 10,
  },
  button: {
    width: '45%',
  },
  textInput: {
    fontSize: 18,
  }
});
