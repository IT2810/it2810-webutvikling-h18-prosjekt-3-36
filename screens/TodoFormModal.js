import React from "react";
import { StyleSheet, Text, View, Alert, Button, TextInput, Platform } from "react-native";
import HeaderButton from "../components/HeaderButton";
import DatePicker from "react-native-datepicker";
import { Modal } from "react-native";

export default class TodoForm extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      headerTitle: "test",
      headerLeft: <HeaderButton onPress={() => navigation.goBack()} title="Cancel" />,
      headerRight: <HeaderButton onPress={() => navigation.add()} disabled={true} title="Add" />
    };
  };

  constructor(props) {
    super(props);
    this.state = {
      helpAlert: true
    };
  }

  componentWillMount() {
    this.props.navigation.setParams({ add: this.add });
  }

  add = () => {
    console.log("save");
  };

  addNewTodo = () => {
    // Kjør addNewTodo metoden til TodoScreen så alt av todo states blir gjort av den.
    if (this.state.helpAlert && this.state.date === undefined) {
      // Om brukern ikke har valgt en dato burde det komme en liten varsling om at den ikke kan bli sett i kalendern.
      Alert.alert(
        "No date chosen",
        "You have not chosen a date and will therefore not be able to view your todo in Calender mode, swipe right to view your todos without date.",
        [{ text: "Never remind me again", onPress: () => this.setState({ helpAlert: false }) }, { text: "OK" }],
        { cancelable: false }
      );
    }
    this.props.addNewTodo(this.state.text, this.state.date);
    this.props.toggleModal();
    this.clearForm();
  };

  clearForm = () => {
    this.setState({ text: undefined });
    this.setState({ date: undefined });
  };

  render() {
    return (
      <View style={{ flex: 1, backgroundColor: "white" }}>
        <TextInput
          editable={true}
          multiline
          style={{ minHeight: 200 }}
          onChangeText={text => this.setState({ text })}
          value={this.state.text}
          placeholder="Write something..."
          placeholderTextColor="grey"
          autoFocus={true}
          underlineColorAndroid="rgba(0,0,0,0)"
          onSubmitEditing={this.addNewTodo}
        />
        <DatePicker
          style={{ width: 200 }}
          date={this.state.date}
          mode="date"
          placeholder="select date"
          format="YYYY-MM-DD"
          minDate="1990-01-01"
          maxDate="2025-01-01"
          confirmBtnText="Confirm"
          cancelBtnText="Cancel"
          customStyles={{
            dateIcon: {
              position: "absolute",
              left: 0,
              top: 4,
              marginLeft: 0
            },
            dateInput: {
              marginLeft: 36
            }
          }}
          onDateChange={date => {
            this.setState({ date: date });
          }}
        />
        <Text>Add a date for your todo (optional)</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  buttonContainer: {
    flex: 1,
    maxHeight: 50,
    flexDirection: "row",
    justifyContent: "space-around",
    padding: 10
  },
  button: {
    width: "45%"
  },
  textInput: {
    fontSize: 18
  }
});
