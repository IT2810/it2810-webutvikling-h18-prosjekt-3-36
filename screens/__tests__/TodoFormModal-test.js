import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoFormModal from "../TodoFormModal";
import appState from '../../store/reducers/Reducer'
import { createStore } from 'redux';

const store = createStore(appState)

it("TodoFormModal Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<TodoFormModal store={store} navigation={{ getParam: () => { }, setParams: () => { } }} isModalVisible={false} />);

    const renderer2 = new ShallowRenderer();
    renderer2.render(renderer.getRenderOutput());
    const result = renderer2.getRenderOutput();
    expect(result).toMatchSnapshot();
});