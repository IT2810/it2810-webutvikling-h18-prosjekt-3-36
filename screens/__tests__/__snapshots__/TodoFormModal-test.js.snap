// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFormModal Matches snapshot. 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#efeff4",
      "flex": 1,
      "height": "100%",
    }
  }
>
  <DatePicker
    TouchableComponent={[Function]}
    allowFontScaling={true}
    androidMode="default"
    cancelBtnText="Cancel"
    confirmBtnText="Confirm"
    customStyles={
      Object {
        "btnTextCancel": Object {
          "color": "#ff2d55",
        },
        "btnTextConfirm": Object {
          "color": "#ff2d55",
        },
      }
    }
    date=""
    disabled={false}
    duration={300}
    format="YYYY-MM-DD"
    height={259}
    hideText={true}
    iconSource={1}
    maxDate="2025-01-01"
    minDate="1990-01-01"
    modalOnResponderTerminationRequest={[Function]}
    mode="date"
    onDateChange={[Function]}
    placeholder="Click here to select date"
    showIcon={false}
    style={
      Object {
        "height": 0,
        "width": 0,
      }
    }
  />
  <TableView
    style={
      Object {
        "backgroundColor": "#efeff4",
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <Section
      allowFontScaling={true}
      footer={null}
      footerComponent={null}
      footerTextColor="#6d6d72"
      header={null}
      headerComponent={null}
      headerTextColor="#6D6D72"
      hideSeparator={false}
      sectionPaddingBottom={15}
      sectionPaddingTop={15}
      sectionTintColor="#EFEFF4"
      separatorInsetLeft={15}
      separatorInsetRight={0}
      separatorTintColor="#C8C7CC"
    >
      <Cell
        accessory={false}
        accessoryColor="#007AFF"
        accessoryColorDisclosureIndicator="#C7C7CC"
        allowFontScaling={true}
        backgroundColor="#FFF"
        cellAccessoryView={null}
        cellContentView={
          <TextInput
            allowFontScaling={true}
            autoFocus={true}
            onChangeText={[Function]}
            placeholder="Title"
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
          />
        }
        cellImageView={null}
        cellStyle="Basic"
        contentContainerStyle={Object {}}
        detail=""
        detailTextStyle={Object {}}
        disableImageResize={false}
        hideSeparator={false}
        highlightActiveOpacity={0.8}
        highlightUnderlayColor="black"
        image={null}
        isDisabled={false}
        leftDetailColor="#007AFF"
        onHighlightRow={null}
        onPress={false}
        onUnHighlightRow={null}
        rightDetailColor="#8E8E93"
        subtitleColor="#000"
        subtitleTextStyle={Object {}}
        title=""
        titleTextColor="#000"
        titleTextStyle={Object {}}
        titleTextStyleDisabled={Object {}}
      />
      <RNPickerSelect
        animationType="slide"
        disabled={false}
        doneText="Done"
        hideDoneBar={false}
        hideIcon={false}
        itemKey={null}
        items={
          Array [
            Object {
              "label": "Work",
              "value": "Work",
            },
            Object {
              "label": "Fun",
              "value": "Fun",
            },
            Object {
              "label": "School",
              "value": "School",
            },
            Object {
              "label": "Other",
              "value": "Other",
            },
          ]
        }
        mode="dialog"
        onDonePress={null}
        onDownArrow={null}
        onUpArrow={null}
        onValueChange={[Function]}
        placeholder={Object {}}
        placeholderTextColor="#C7C7CD"
        style={Object {}}
        value="Work"
      >
        <Cell
          accessory={false}
          accessoryColor="#007AFF"
          accessoryColorDisclosureIndicator="#C7C7CC"
          allowFontScaling={true}
          backgroundColor="#FFF"
          cellAccessoryView={null}
          cellContentView={null}
          cellImageView={null}
          cellStyle="RightDetail"
          contentContainerStyle={Object {}}
          detail="Work"
          detailTextStyle={Object {}}
          disableImageResize={false}
          hideSeparator={false}
          highlightActiveOpacity={0.8}
          highlightUnderlayColor="black"
          image={null}
          isDisabled={false}
          leftDetailColor="#007AFF"
          onHighlightRow={null}
          onPress={false}
          onUnHighlightRow={null}
          rightDetailColor="#8E8E93"
          subtitleColor="#000"
          subtitleTextStyle={Object {}}
          title="Category"
          titleTextColor="#000"
          titleTextStyle={Object {}}
          titleTextStyleDisabled={Object {}}
        />
      </RNPickerSelect>
    </Section>
    <Section
      allowFontScaling={true}
      footer={null}
      footerComponent={null}
      footerTextColor="#6d6d72"
      header={null}
      headerComponent={null}
      headerTextColor="#6D6D72"
      hideSeparator={false}
      sectionPaddingBottom={15}
      sectionPaddingTop={15}
      sectionTintColor="#EFEFF4"
      separatorInsetLeft={15}
      separatorInsetRight={0}
      separatorTintColor="#C8C7CC"
    >
      <Cell
        accessory={false}
        accessoryColor="#007AFF"
        accessoryColorDisclosureIndicator="#C7C7CC"
        allowFontScaling={true}
        backgroundColor="#FFF"
        cellAccessoryView={null}
        cellContentView={null}
        cellImageView={null}
        cellStyle="RightDetail"
        contentContainerStyle={Object {}}
        detail="Select"
        detailTextStyle={Object {}}
        disableImageResize={false}
        hideSeparator={false}
        highlightActiveOpacity={0.8}
        highlightUnderlayColor="black"
        image={null}
        isDisabled={false}
        leftDetailColor="#007AFF"
        onHighlightRow={null}
        onPress={[Function]}
        onUnHighlightRow={null}
        rightDetailColor="#8E8E93"
        subtitleColor="#000"
        subtitleTextStyle={Object {}}
        title="Date"
        titleTextColor="#000"
        titleTextStyle={Object {}}
        titleTextStyleDisabled={Object {}}
      />
      <Cell
        accessory="DisclosureIndicator"
        accessoryColor="#007AFF"
        accessoryColorDisclosureIndicator="#C7C7CC"
        allowFontScaling={true}
        backgroundColor="#FFF"
        cellAccessoryView={null}
        cellContentView={null}
        cellImageView={null}
        cellStyle="RightDetail"
        contentContainerStyle={Object {}}
        detail=" Select"
        detailTextStyle={Object {}}
        disableImageResize={false}
        hideSeparator={false}
        highlightActiveOpacity={0.8}
        highlightUnderlayColor="black"
        image={null}
        isDisabled={false}
        leftDetailColor="#007AFF"
        onHighlightRow={null}
        onPress={[Function]}
        onUnHighlightRow={null}
        rightDetailColor="#8E8E93"
        subtitleColor="#000"
        subtitleTextStyle={Object {}}
        title="Location"
        titleTextColor="#000"
        titleTextStyle={Object {}}
        titleTextStyleDisabled={Object {}}
      />
    </Section>
  </TableView>
</Component>
`;
